### ISSUE TITLE
Xcode 15 beta and Firestore 10.10: The signature of “gRPC-C++.xcframework” cannot be verified

### ISSUE THREAD

**Issue by ConfuseIous** [firebase/firebase-ios-sdk#11401]
### Description

In Xcode 15 beta, updating to Firebase 10.10 throws a build error. "The signature of “gRPC-C++.xcframework” cannot be verified."

The full error is:
```
/Users/username/Library/Developer/Xcode/DerivedData/OptimisingFirestore-ejvnmdkwtmsvjjbnruaonpkfzunf/SourcePackages/artifacts/grpc-binary/gRPC-C++/gRPC-C++.xcframework:1:1: error: The signature of “gRPC-C++.xcframework” cannot be verified.
    note: Codesign verification failed
    note: /Users/username/Library/Developer/Xcode/DerivedData/OptimisingFirestore-ejvnmdkwtmsvjjbnruaonpkfzunf/SourcePackages/artifacts/grpc-binary/gRPC-C++/gRPC-C++.xcframework: bundle format unrecognized, invalid, or unsuitable
```

This issue does not occur when using Xcode 14.3, even when building for iOS 17 targets. As a temporary fix, downgrading to Firebase 10.8.1 resolves the issue.

### Reproducing the issue

1. Download Xcode 15
2. Download the [sample project](https://github.com/ConfuseIous/OptimisingFirestore)  (or just create a new Xcode project in Xcode 15 and install Firestore)
3. Observe that there are no issues when building
4. Update Firebase to 10.10
5. Observe that the error listed above is thrown

### Firebase SDK Version

10.10

### Xcode Version

15

### Installation Method

Swift Package Manager

### Firebase Product(s)

Firestore

### Targeted Platforms

iOS

### Relevant Log Output

```shell
/Users/username/Library/Developer/Xcode/DerivedData/OptimisingFirestore-ejvnmdkwtmsvjjbnruaonpkfzunf/SourcePackages/artifacts/grpc-binary/gRPC-C++/gRPC-C++.xcframework:1:1: error: The signature of “gRPC-C++.xcframework” cannot be verified.
    note: Codesign verification failed
    note: /Users/username/Library/Developer/Xcode/DerivedData/OptimisingFirestore-ejvnmdkwtmsvjjbnruaonpkfzunf/SourcePackages/artifacts/grpc-binary/gRPC-C++/gRPC-C++.xcframework: bundle format unrecognized, invalid, or unsuitable
```

### If using Swift Package Manager, the project's Package.resolved

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```yml

{
  "pins" : [
    {
      "identity" : "abseil-cpp-binary",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/abseil-cpp-binary.git",
      "state" : {
        "revision" : "bfc0b6f81adc06ce5121eb23f628473638d67c5c",
        "version" : "1.2022062300.0"
      }
    },
    {
      "identity" : "firebase-ios-sdk",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/firebase/firebase-ios-sdk",
      "state" : {
        "revision" : "5034479ca0c4f32f299677fb0ba1fe4d3e3e20b4",
        "version" : "10.10.0"
      }
    },
    {
      "identity" : "googleappmeasurement",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/GoogleAppMeasurement.git",
      "state" : {
        "revision" : "69f5b8425dd473d2f2475bba7f88f82cd9da0e36",
        "version" : "10.10.0"
      }
    },
    {
      "identity" : "googledatatransport",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/GoogleDataTransport.git",
      "state" : {
        "revision" : "7874c1b48cbffd086ce8a052c4be873a78613775",
        "version" : "9.2.3"
      }
    },
    {
      "identity" : "googleutilities",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/GoogleUtilities.git",
      "state" : {
        "revision" : "871d43135925cde39ef7421d8723ce47edfdcc39",
        "version" : "7.11.1"
      }
    },
    {
      "identity" : "grpc-binary",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/grpc-binary.git",
      "state" : {
        "revision" : "2b7656d8ee892cd39db460b1a0d849de618405e6",
        "version" : "1.50.1"
      }
    },
    {
      "identity" : "gtm-session-fetcher",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/gtm-session-fetcher.git",
      "state" : {
        "revision" : "d415594121c9e8a4f9d79cecee0965cf35e74dbd",
        "version" : "3.1.1"
      }
    },
    {
      "identity" : "leveldb",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/firebase/leveldb.git",
      "state" : {
        "revision" : "0706abcc6b0bd9cedfbb015ba840e4a780b5159b",
        "version" : "1.22.2"
      }
    },
    {
      "identity" : "nanopb",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/firebase/nanopb.git",
      "state" : {
        "revision" : "819d0a2173aff699fb8c364b6fb906f7cdb1a692",
        "version" : "2.30909.0"
      }
    },
    {
      "identity" : "promises",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/promises.git",
      "state" : {
        "revision" : "ec957ccddbcc710ccc64c9dcbd4c7006fcf8b73a",
        "version" : "2.2.0"
      }
    },
    {
      "identity" : "swift-protobuf",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/apple/swift-protobuf.git",
      "state" : {
        "revision" : "f25867a208f459d3c5a06935dceb9083b11cd539",
        "version" : "1.22.0"
      }
    }
  ],
  "version" : 2
}


```

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

Replace this line with the contents of your Podfile.lock!

```

</details>

**Comment by paulb777** [comment]
In the meantime, a workaround would be to go back to the Firestore source distribution in Firebase 10.7.0 or earlier.

**Comment by ncooke3** [comment]
Hi everyone, this issue should be fixed with a newly released gRPC version. Please update your package dependencies to pull this in.

**Comment by TuncayEnsi** [comment]
this fixed the issue, but opened a new segmentation problem:I get this compiler error when deploying "Unable to execute command: Segmentation fault: 11" and "Linker command failed due to signal (use -v to see invocation)"

**Comment by TuncayEnsi** [comment]
(Xcode 15 Beta of course)

**Comment by ncooke3** [comment]
Hmm, I'm not reproducing that @TuncayEnsi in my sample app. What device/OS are deploying to? Are you deploying to the simulator?

**Comment by TuncayEnsi** [comment]
No to a physical device, I am not able to. I am not getting any error when previewing in swiftUI thou...

**Comment by TuncayEnsi** [comment]
iPhone 12 Pro Max

**Comment by ncooke3** [comment]
@TuncayEnsi, I'm not too sure what is going on. I was able to successfully deploy to an iPhone XR running iOS 16.4.

**Comment by ncooke3** [comment]
Could this be related to your project's bitcode configuration? See [this](https://stackoverflow.com/a/41454965/9331576) SO answer. I'm trying to understand why it's throwing the segmentation fault..

**Comment by TuncayEnsi** [comment]
Thanks for the answer!

I figured it had something to do with that. But I cannot find this option as shown in Stackoverflow in Xcode 15 any longer.
Therefore I cannot turn that of.

Is this something caused by me or has apple moved that?

**Comment by TuncayEnsi** [comment]
Deploying to iOS 17 Beta

**Comment by TuncayEnsi** [comment]
> 

@ncooke3 yes deleted all those files; also cleaned build folder via Xcode several times, still running into this

**Comment by JamieShaneRainbird** [comment]
Hi everyone, 
Not sure if others are still experiencing the issue & I'm going to be honest -- I am not at all a pro when it comes to Xcode but, after the whole day of trying different things and suggestions I managed to resolve the issue by:

1. Removing the Package Dependency
2. Clean Build Folder...
3. Quitting and relaunching Xcode
4. Reintalling [https://github.com/firebase/firebase-ios-sdk] to "Up to Next Major Version" 10.10.0 < 11.0.0
5. Going into "Xcode" in the menu bar and pressing "settings"
6. Going to the Locations 
7. Re-configure the Command Line Tools 

No idea if they're fully connected together but, it seemed to have work & my app is running as it was with firebase and no more errors are appearing.

**Comment by TuncayEnsi** [comment]
thanks for the insight, for me nothing changed thou still not able to deploy to a real device

**Comment by TuncayEnsi** [comment]
I read a lot of articles saying that bit coding is deprecated since Xcode 14

**Comment by ncooke3** [comment]
@TuncayEnsi, I'm not sure if there is anyway to configure bit code at this point. See this [note](https://developer.apple.com/documentation/xcode-release-notes/xcode-15-release-notes#Deprecations). This makes me think the issue is something else. But I'm not sure if it's Firebase related–– did you try building and deploying your app with an older version of Firebase like 10.7.0 (see @paulb777's [comment](https://github.com/firebase/firebase-ios-sdk/issues/11401#issuecomment-1578881801) above)?

**Comment by TuncayEnsi** [comment]
Build and deploying works and everything working fine on Firebase 10.7.0, I guess the problem is firebase therefore?

**Comment by TuncayEnsi** [comment]
I will Stick to that version as it deploys on iOS 17 until this gets fixed. Is there anything bad about doing that?

**Comment by TuncayEnsi** [comment]
I am also wondering why gRPC is not throwing the same signing issue in this version as it has also been downgraded

**Comment by ncooke3** [comment]
> I will Stick to that version as it deploys on iOS 17 until this gets fixed. Is there anything bad about doing that?

Nothing _bad_, but there are some various bug fixes and features that 10.7.0 does not include. With regard to Firestore, 10.8.0 and above gives much faster build times for Firestore.

> I am also wondering why gRPC is not throwing the same signing issue in this version as it has also been downgraded

gRPC is not throwing the same signing issue in this version likely because it is being built from source rather than a binary. In Firebase 10.8.0+, gRPC is included as a pre-compiled xcframework. This xcframework has a certificates bundle in the xcframework's `Resources` directory. In Xcode 15, such a bundle that's part of a binary distribution needs to be signed accordingly. In our case, this certificates bundle isn't used so I removed it in the latest gRPC release that I released today– which fixed the signing issue as signing was no longer needed.

---

Sincere apologies for the trouble here @TuncayEnsi, and thank you for helping narrow things down. It seems that Firebase 10.8.0+ is the issue, but I still don't know why. I will continue looking into it. If you are able to provide a minimum reproducible example project or error messages with any greater level of detail– that'd be greatly appreciated.
